src/acl/external/time_quota/ext_time_quota_acl.cc

```diff
@@ -72,141 +68,106 @@ TDB_CONTEXT *db = nullptr;
  * same activity and the quota will be reduced.
  */
 static int pauseLength = 300;
+static bool noClearDb = false;
 
-static FILE *logfile = stderr;
+static std::ostream *logfile = &std::cerr;
 static int tq_debug_enabled = false;
 
 static void open_log(const char *logfilename)
 {
-    logfile = fopen(logfilename, "a");
-    if ( logfile == NULL ) {
+    static auto log = new std::ofstream(logfilename, std::ios::app);
+    if (log && *log)
+        logfile = log;
+    else
         perror(logfilename);
-        logfile = stderr;
-    }
-}
-
-static void vlog(const char *level, const char *format, va_list args)
-{
-    time_t now = time(NULL);
-
-    fprintf(logfile, "%ld %s| %s: ", static_cast<long int>(now),
-            program_name, level);
-    vfprintf (logfile, format, args);
-    fflush(logfile);
 }
 
-static void log_debug(const char *format, ...)
-{
-    va_list args;
-
-    if ( tq_debug_enabled ) {
-        va_start (args, format);
-        vlog("DEBUG", format, args);
-        va_end (args);
+#define log_debug(CONTENT) \
+    if (tq_debug_enabled) { \
+         *logfile << \
+            std::chrono::system_clock::to_time_t(std::chrono::system_clock::now()) << \
+             " DEBUG| " << CONTENT << std::endl ;\
     }
-}
 
-static void log_info(const char *format, ...)
-{
-    va_list args;
-
-    va_start (args, format);
-    vlog("INFO", format, args);
-    va_end (args);
-}
-
-static void log_error(const char *format, ...)
-{
-    va_list args;
+#define low_log(LEVEL, CONTENT) \
```

Comment:
Please try to use debugs() instead. We have been upgrading primary code to allow debugs() usage in helpers. More work may be needed, but at least some helpers were converted already IIRC, so this is worth trying (instead of adding more macros). Find commits that add debugs() logging to helpers and use them as blueprints for how to set things up at helper startup time.

Reply:
I'm not sure how helpers have used debugs() so far; I had to add some code to it to not crash on startup